modules/unsupported/mongodb/src/main/java/org/geotools/data/mongodb/FilterToMongo.java

```diff
@@ -621,6 +621,8 @@ Object encodeLiteral(Object literal, Class<?> targetType) {
             }
             // by default, return string as is
             return literal;
+        } else if (literal instanceof Number) {
+            return literal;
```

Comment:
Good point!

Looks like: 

double
32-bit integer
64-bit integer
decimal

From: https://docs.mongodb.com/manual/reference/operator/query/type/

Might be better suited to check for Integer, Double, and Long only.   Thoughts?